relay-profiling/src/utils.rs

```diff
@@ -16,28 +16,34 @@ where
 {
     #[derive(Deserialize)]
     #[serde(untagged)]
-    enum StringOrNumber<T> {
-        String(String),
-        Number(T),
-        Bool(bool),
+    enum AnyType<T> {
         Array(Vec<Value>),
+        Bool(bool),
+        Null,
+        Number(T),
         Object(Map<String, Value>),
+        String(String),
     }
 
-    match StringOrNumber::<T>::deserialize(deserializer)? {
-        StringOrNumber::String(s) => s.parse::<T>().map_err(serde::de::Error::custom),
-        StringOrNumber::Number(n) => Ok(n),
-        StringOrNumber::Bool(v) => Err(serde::de::Error::custom(format!(
+    match AnyType::<T>::deserialize(deserializer)? {
+        AnyType::String(s) => s.parse::<T>().map_err(serde::de::Error::custom),
+        AnyType::Number(n) => Ok(n),
+        AnyType::Bool(v) => Err(serde::de::Error::custom(format!(
             "unsupported value: {:?}",
             v
         ))),
-        StringOrNumber::Array(v) => Err(serde::de::Error::custom(format!(
+        AnyType::Array(v) => Err(serde::de::Error::custom(format!(
             "unsupported value: {:?}",
             v
         ))),
-        StringOrNumber::Object(v) => Err(serde::de::Error::custom(format!(
+        AnyType::Object(v) => Err(serde::de::Error::custom(format!(
             "unsupported value: {:?}",
             v
         ))),
+        AnyType::Null => Err(serde::de::Error::custom("unsupported null value")),
     }
 }
+
+pub fn string_is_null_or_empty(s: &Option<String>) -> bool {
+    s.as_ref().unwrap_or(&"".to_string()).is_empty()
+}
```

Comment:
This avoids allocating an extra string:
```suggestion
pub fn string_is_null_or_empty(s: &Option<String>) -> bool {
    s.as_ref().map(|s| s.is_empty()).unwrap_or(true)
}
```
Generally, it's more convenient to work with `&Option<>` vs `Option<&>`, so updating the arg would be great!

Reply:
This can be nicely expressed with `map_or`:

```rs
pub fn string_is_null_or_empty(s: &Option<String>) -> bool {
    s.as_deref().map_or(true, |s| s.is_empty())
}
```